Exception in prawn when generating a PDF from IDE

It looking at the exception, it looks like an issue with prawn and syntax highlighting: `prawn.text.formatted.wrap`

I locked the trigger down to the quick-start.adoc file and reduced it to the following snippet that fails with the PDF generation in the IDE. 

```
[source,java]
----
                    .exchange("http://" + Constants.REST_COMPUTE_SERVICE + "/compute", HttpMethod.PUT)
----
```

The project you referenced doesn't include a PDF generation, therefore I am not sure what versions you use to create a PDF.

The IntelliJ plugin uses asciidoctorj-pdf:1.5.0-beta.4 together with the latest AsciiDoctorJ runtime. 

Please use the latest versions of asciidoctorj and asciidoctorj-pdf in your maven build. If you see the same problem there, please raise an issue with asciidoctor-pdf with a minimal example (usually not asciidoctorj-pdf, as this is just a re-packaging). 

Once you have a working Maven build with the latest versions and still have a failing IntelliJ plugin, please raise an issue with the plugin.

_Originally posted by @ahus1 in https://github.com/asciidoctor/asciidoctor-intellij-plugin/issues/325#issuecomment-529922968_

@mojavelinux - it is happening with asciidoctorj-pdf beta.5. Something in prawn. I copy the JRuby stack trace here. Wasn't able to reproduce it on the command line, but the JRuby command line seems to use a different version. Seems to happen once the verbatim content has an automatic line break, but not without the line break.

I'm about to create a minimal test case for it. 

```
Invalid start: 1
java.lang.AssertionError: Invalid start: 1
	at org.jruby.util.ByteList.append(ByteList.java:542)
	at org.jruby.util.ByteList.dup(ByteList.java:330)
	at org.jruby.util.ByteList.dup(ByteList.java:300)
	at org.jruby.RubyString.modify(RubyString.java:965)
	at org.jruby.RubyString.modify19(RubyString.java:976)
	at org.jruby.RubyString.force_encoding(RubyString.java:6089)
	at org.jruby.RubyString.force_encoding(RubyString.java:6084)
	at org.jruby.RubyString$INVOKER$i$1$0$force_encoding.call(RubyString$INVOKER$i$1$0$force_encoding.gen)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:339)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:170)
	at uri_3a_classloader_3a_.gems.prawn_minus_2_dot_2_dot_2.lib.prawn.text.formatted.wrap.invokeOther0:force_encoding(uri:classloader:/gems/prawn-2.2.2/lib/prawn/text/formatted/wrap.rb:106)
	at uri_3a_classloader_3a_.gems.prawn_minus_2_dot_2_dot_2.lib.prawn.text.formatted.wrap.RUBY$block$print_line$0(uri:classloader:/gems/prawn-2.2.2/lib/prawn/text/formatted/wrap.rb:106)
	at org.jruby.runtime.CompiledIRBlockBody.yieldDirect(CompiledIRBlockBody.java:146)
	at org.jruby.runtime.MixedModeIRBlockBody.yieldDirect(MixedModeIRBlockBody.java:124)
	at org.jruby.runtime.BlockBody.yield(BlockBody.java:114)
	at org.jruby.runtime.Block.yield(Block.java:165)
	at org.jruby.RubyArray.collect(RubyArray.java:2563)
	at org.jruby.RubyArray.map19(RubyArray.java:2577)
	at org.jruby.RubyArray$INVOKER$i$0$0$map19.call(RubyArray$INVOKER$i$0$0$map19.gen)
	at org.jruby.internal.runtime.methods.JavaMethod$JavaMethodZeroBlock.call(JavaMethod.java:555)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:80)
	at org.jruby.runtime.callsite.CachingCallSite.callIter(CachingCallSite.java:89)
	at org.jruby.ir.instructions.CallBase.interpret(CallBase.java:537)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:362)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:80)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:138)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:125)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:192)
	at org.jruby.runtime.callsite.CachingCallSite.cacheAndCall(CachingCallSite.java:318)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:139)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:346)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:105)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:92)
	at org.jruby.ir.runtime.IRRuntimeHelpers.unresolvedSuper(IRRuntimeHelpers.java:1191)
	at org.jruby.ir.instructions.UnresolvedSuperInstr.interpret(UnresolvedSuperInstr.java:102)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:362)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.ir.interpreter.InterpreterEngine.interpret(InterpreterEngine.java:86)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:171)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:158)
	at org.jruby.internal.runtime.methods.DynamicMethod.call(DynamicMethod.java:200)
	at org.jruby.runtime.callsite.CachingCallSite.call(CachingCallSite.java:168)
	at org.jruby.ir.interpreter.InterpreterEngine.processCall(InterpreterEngine.java:317)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:72)
	at org.jruby.ir.interpreter.Interpreter.INTERPRET_BLOCK(Interpreter.java:128)
	at org.jruby.runtime.MixedModeIRBlockBody.commonYieldPath(MixedModeIRBlockBody.java:151)
	at org.jruby.runtime.IRBlockBody.yieldSpecific(IRBlockBody.java:89)
	at org.jruby.runtime.Block.yieldSpecific(Block.java:134)
	at org.jruby.ir.runtime.IRRuntimeHelpers.yieldSpecific(IRRuntimeHelpers.java:482)
	at org.jruby.ir.instructions.YieldInstr.interpret(YieldInstr.java:76)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.processOtherOp(StartupInterpreterEngine.java:178)
	at org.jruby.ir.interpreter.StartupInterpreterEngine.interpret(StartupInterpreterEngine.java:104)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.INTERPRET_METHOD(MixedModeIRMethod.java:105)
	at org.jruby.internal.runtime.methods.MixedModeIRMethod.call(MixedModeIRMethod.java:92)
	at org.jruby.ir.runtime.IRRuntimeHelpers.unresolvedSuper(IRRuntimeHelpers.java:1191)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_pdf_minus_1_dot_5_dot_0_dot_beta_dot_5.lib.asciidoctor_minus_pdf.prawn_ext.extensions.invokeSuper30:-unknown-super-target-(uri:classloader:/gems/asciidoctor-pdf-1.5.0.beta.5/lib/asciidoctor-pdf/prawn_ext/extensions.rb:249)
	at uri_3a_classloader_3a_.gems.asciidoctor_minus_pdf_minus_1_dot_5_dot_0_dot_beta_dot_5.lib.asciidoctor_minus_pdf.prawn_ext.extensions.RUBY$method$font_size$0(uri:classloader:/gems/asciidoctor-pdf-1.5.0.beta.5/lib/asciidoctor-pdf/prawn_ext/extensions.rb:249)
```

--- a/CHANGELOG.adoc
+++ b/CHANGELOG.adoc
@@ -10,6 +10,7 @@ This document provides a high-level view of the changes introduced by release.
 === 0.30.8 (work in progress)
 
 - highlight warnings for Spring REST Docs at line in editor
+- upgrade to JRuby 9.2.8.0 to avoid assertion errors when creating PDFs (#337)
 
 === 0.30.7
 

--- a/build.gradle
+++ b/build.gradle
@@ -96,6 +96,7 @@ test {
 dependencies {
   compile 'org.asciidoctor:asciidoctorj-diagram:1.5.18'
   compile 'org.asciidoctor:asciidoctorj-pdf:1.5.0-beta.5'
+  compile 'org.jruby:jruby:9.2.8.0' // TODO: remove JRuby dependency once asciidoctorj has upgraded
   /* snakeyaml is s used by asciidoctorj-pdf, but is actually provided within jruby-stdlib
    * a snakeyaml version in the classpath takes precedence, but IntelliJ includes a version that is too old
    * therefore this plugin includes the same version of snakeyaml that is already included in jruby-stdlib

